frontend/src/components/common/InviteButton.vue

```diff
@@ -0,0 +1,228 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { computed, ref, onMounted } from 'vue';
+
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import ShareLinkContent from '@/components/object/share/ShareLinkContent.vue';
+import { Button, Dialog, TabView, TabPanel, RadioButton, InputText, useToast, InputSwitch } from '@/lib/primevue';
+import { useConfigStore, useObjectStore, useBucketStore } from '@/store';
+import { permissionService } from '@/services';
+
+import type { Ref } from 'vue';
+import type { COMSObject, Bucket } from '@/types';
+
+// Props
+type Props = {
+  bucketId?: string;
+  objectId?: string;
+  labelText: string;
+  restricted?: boolean;
+};
+export type inviteFormData = {
+  email?: string;
+  bucketId?: string;
+  expiresAt?: number;
+  objectId?: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  bucketId: '',
+  objectId: '',
+  labelText: '',
+  isRestricted: false
+});
+
+// Store
+const objectStore = useObjectStore();
+const bucketStore = useBucketStore();
+const { getConfig } = storeToRefs(useConfigStore());
+const toast = useToast();
+
+// State
+const obj: Ref<COMSObject | undefined> = ref(undefined);
+const bucket: Ref<Bucket | undefined> = ref(undefined);
+const isRestricted: Ref<boolean> = ref(props.restricted);
+const showInviteLink: Ref<boolean> = ref(false);
+
+// Share link
+const inviteLink: Ref<string> = ref('');
+
+// Value is set in hours, so 3 days (3x24) = 72hours
+const timeFrames: Ref<
+  {
+    name: string;
+    value: number;
+  }[]
+> = ref([
+  { name: '1 Hour', value: 3600 },
+  { name: '8 Hour', value: 28800 },
+  { name: '1 Day (Default)', value: 86400 },
+  { name: '1 Week', value: 691199 }
+]);
+const formData: Ref<inviteFormData> = ref({ expiresAt: 86400 });
+
+// Dialog
+const displayInviteDialog = ref(false);
+
+// Share link
+const bcBoxLink = computed(() => {
+  const path = props.objectId ? `detail/objects?objectId=${props.objectId}` : `list/objects?bucketId=${props.bucketId}`;
+  return `${window.location.origin}/${path}`;
+});
+const comsUrl = computed(() => {
+  return `${getConfig.value.coms?.apiPath}/object/${props.objectId}`;
+});
```

Comment:
This shouldn't be something we need to be caching in the component - this looks like something that should be residing in the Service layer. Not completely sure but on quick glance this invite button shouldn't be needing to send a direct link into coms on Line 136?